source/reference/minio-server/settings/storage-class.rst

```diff
@@ -23,80 +34,140 @@ This impacts how MinIO uses the space on the drive(s) and how MinIO can recover
-   MinIO shards objects written with the default ``STANDARD`` storage class using this parity value.
+The :ref:`parity level <minio-ec-parity>` for objects written with the ``REDUCED`` storage class.
 
-   MinIO references the ``x-amz-storage-class`` header in request metadata for determining which storage class to assign an object. 
-   The specific syntax or method for setting headers depends on your preferred method for interfacing with the MinIO server.
+MinIO references the ``x-amz-storage-class`` header in request metadata for determining which storage class to assign an object. 
+The specific syntax or method for setting headers depends on your preferred method for interfacing with the MinIO server.
 
-   Specify the value using ``EC:M`` notation, where ``M`` refers to the number of parity blocks to create for the object.
+Specify the value using ``EC:M`` notation, where ``M`` refers to the number of parity blocks to create for the object.
 
-   The following table lists the default values based on the :ref:`erasure set size <minio-ec-erasure-set>` of the initial server pool in the deployment:
+This value **must be** less than or equal to :envvar:`MINIO_STORAGE_CLASS_STANDARD`.
 
-   .. list-table::
-      :header-rows: 1
-      :widths: 30 70
-      :width: 100%
+You cannot set this value for deployments with an erasure set size less than 2.
+Defaults to ``EC:1`` for deployments with erasure set size greater than 1.
+Defaults to ``EC:0`` for deployments of erasure set size of 1.
 
-      * - Erasure Set Size
-        - Default Parity (EC:N)
+Parity Retention Optimization
+-----------------------------
 
-      * - 1
-        - EC:0
 
-      * - 2-3
-        - EC:1
+.. tab-set::
 
-      * - 4-5
-        - EC:2
+   .. tab-item:: Environment Variable
+      :sync: envvar
 
-      * - 6 - 7
-        - EC:3
+      .. envvar:: MINIO_STORAGE_CLASS_OPTIMIZE
 
-      * - 8 - 16
-        - EC:4
+   .. tab-item:: Configuration Setting
+      :sync: config
 
-   The minimum supported value is ``0``, which indicates no erasure coding protections.
-   These deployments rely entirely on the storage controller or resource for availability / resiliency. 
-   
-   The maximum value depends on the erasure set size of the initial server pool in the deployment, where the upper bound is  :math:`\frac{\text{ERASURE_SET_SIZE}}{\text{2}}`.
-   For example, a deployment with erasure set stripe size of 16 has a maximum standard parity of 8.
+      .. mc-conf:: storage_class optimize
+         :delimiter: " "
 
-   You can change this value after startup to any value between ``0`` and the upper bound for the erasure set size.
-   MinIO only applies the changed parity to newly written objects.
-   Existing objects retain the parity value in place at the time of their creation.
+MinIO by default automatically "upgrades" parity for an object if the destination erasure set maintains write quorum *but* has one or more drives offline.
```

Comment:
```suggestion
MinIO by default automatically "upgrades" parity for an object if the destination erasure set maintains write quorum *but* has one or more drives offline.
The object written on the destination maintains the same number of data shards, but a reduced number of parity shards compared to the original.
```

I think we need an example to clarify how MinIO "upgrades" parity.
I might have gotten it backwards in my suggestion.

Reply:
I think so as well - @harshavardhana or perhaps @klauspost can confirm but:

in an erasure set of 16 drives and EC:4, if 1 drive goes online, we write with EC:5.

so the new object is 11:5 while other objects are 12:4. Since a drive is offline, we get  11 data blocks and 4 parity blocks, but the object is maintaining the 'quorum' of all other objects.

If you were to set this for capacity mode, we would write 12:4 and just leave off the last parity block, such that this object has reduced quorum (just like all the other objects on that set). 

AFAIK healing does not correct parity upgrades, though it would 'fix' the object written in capacity mode.

Reply:
https://github.com/minio/minio/blob/f5e3eedf3458c9f5b580edac643292213c1322e7/cmd/erasure-object.go#L1293-L1320

Also implies we upgrade parity for each 'down' drive up to quorum.

I assume the issue is that the smaller number of data shards means those shards are larger, therefore there are some drives that end up hot-spotted if the erasure set operates in it's degraded state for a long time.

Reply:
@djwfyi i might fire this off now and return to parity upgrade behavior in a dedicated PR, since it's likely nuanced.